src/windshaft-integration/model-updater.js

```diff
@@ -130,6 +137,25 @@ ModelUpdater.prototype._generateTorqueTileURLTemplate = function (windshaftMap,
   return windshaftMap.getBaseURL() + '/' + layerIndexes.join(',') + '/{z}/{x}/{y}.json.torque';
 };
 
+ModelUpdater.prototype._updateLegendModels = function (layerModel, remoteLayerIndex, windshaftMap) {
+  var layerMetadata = windshaftMap.getLayerMetadata(remoteLayerIndex);
```

Comment:
Let's review this. This is where rules are converted to legend models, using [these adapters](https://github.com/CartoDB/cartodb.js/tree/d27434a26c552ba40ccd3138e2a7a3f00378f83b/src/windshaft-integration/legends).